testing/test_cache.py

```diff
@@ -87,7 +87,30 @@ def test_cachefuncarg(cache):
         assert result.ret == 0
         result.stdout.fnmatch_lines(["*1 passed*"])
 
-
+    def test_custom_cache_dirname(self, testdir):
+        testdir.makeini("""
+            [pytest]
+            cache_dirname = custom_cache_dirname
+        """)
+        testdir.makepyfile(test_errored='def test_error():\n    assert False')
+        testdir.runpytest()
+        assert os.path.exists('custom_cache_dirname')
+
+    @pytest.mark.parametrize('custom_cache_dirname', [
+        pytest.param('/tmp/mycache', marks=pytest.mark.skipif(sys.platform == 'win32',
```

Comment:
I think you can use just `os.path.abspath('tmp')` and it will work for all platforms, no need to parametrize for each.